### DISCUSSION TITLE
AnalyticalTable first row selected

### DISCUSSION THREAD

**Discussion by francinyreichert** [UI5/webcomponents-react#4258] [Q&A] [answered]
Hello, 

I am using a FlexibleColumnLayout component with the two columns layout and an AnalyticalTable on the first column. I'm trying to make the first row to be selected everytime I enter the application, and if the user select another row this row should be selected:

<img width="1127" alt="image" src="https://user-images.githubusercontent.com/65377988/223138870-3c750688-d63f-4c44-8876-ad9435d1ded4.png">

I tried to do the following:
```
const [selectedRowId, setSelectedRowId] = useState<{ [key: string]: boolean }>({ "0": true });
return (
<AnalyticalTable
         selectedRowIds={selectedRowId} 
         onRowClick={(e?: CustomEvent<{ row?: unknown }>): void => {
            const row = e?.detail.row as Record<string, unknown>;
            setSelectedRowId({ [row.id as string]: true });
          }
        }
/>
);
```

But this didn't work, the first row was not selected. It only worked when I used `selectedRowIds={ "0": true }`, but I can't have this value fixed, I need it to be variable because the user should be able to select other rows. 

Do you have any suggestions on how can I do this?

**Comment by Lukas742** [comment]
Hi @francinyreichert 

the issue you're describing can have multiple root causes, so to be able to help you please provide a reproducible example preferably by leveraging this [codeSandbox](https://codesandbox.io/s/ui5-webcomponents-react-template-29l34) template.

**Comment by francinyreichert** [comment]
Hi @Lukas742, I created [this code sandbox](https://codesandbox.io/s/festive-mclaren-9tb70h?file=/src/App.js) with an example, but it looks like it is working as expected there... I'll have to check my code to try to find the issue.

**Comment by francinyreichert** [comment]
@Lukas742 I tried to reproduce it on codesandbox, but I couldn't. Do you think this issue could be related to multiple re-renders?

**Comment by Lukas742** [comment]
Normally the internal table state should be stable. Did you memoize your `data` array?

**Comment by francinyreichert** [comment]
Apparently this issue is being caused because our table data is being loaded after the table renders...

**Comment by Lukas742** [comment, marked as answer]
The AnalyticalTable resets its internal state when the data changes. You can find out how to prevent that in the "How to stop the table state from automatically resetting when the data changes?" [Recipe](https://sap.github.io/ui5-webcomponents-react/?path=/docs/data-display-analyticaltable-recipes--docs).

**Reply by francinyreichert** [reply]
tested here and it worked! thank you, @Lukas742  :)
